### PR TITLE
Use key in probedPlugin

### PR DESCRIPTION
**Pull request by cwdsuzhou**
**What type of PR is this?**

Since https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/plugins.go#L748 has already used the PluginName as the key in map, so in this loop we could directly the key

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
/cc @thockin 
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/assign @thockin

### CODE REVIEW

File: pkg/volume/plugins.go [hunk 1]
```diff
@@ -714,16 +714,16 @@ func (pm *VolumePluginMgr) FindPluginByName(name string) (VolumePlugin, error) {
 	matchedPluginNames := []string{}
 	matches := []VolumePlugin{}
 	for k, v := range pm.plugins {
-		if v.GetPluginName() == name {
+		if k == name {
```
Comment (JieJhih) [inline pkg/volume/plugins.go]: ``pm.plugins`` type is ``map[string]VolumePlugin``,  so I think the key is unique?
Is this should be cleaner?
```
	if v, ok := pm.plugins[name]; ok {
		matchedPluginNames = append(matchedPluginNames, name)
		matches = append(matches, v)
	}
```

**Review by LexCC — CHANGES_REQUESTED** [review]

File: pkg/volume/plugins.go (hunk 1 above)
Comment (cwdsuzhou) [reply pkg/volume/plugins.go]: Right. thanks, I would change the code.

**Comment by cwdsuzhou** [comment]
/assign @msau42 @jsafrane 

PTAL thanks

File: pkg/volume/plugins.go [hunk 2]
```diff
@@ -713,19 +713,15 @@ func (pm *VolumePluginMgr) FindPluginByName(name string) (VolumePlugin, error) {
 	// Once we can get rid of legacy names we can reduce this to a map lookup.
```
Comment (thockin) [inline pkg/volume/plugins.go]: FWIW - legacy names have been gone a LONG time

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**